File: Python/038.py
from Utils import isPandigital

pandigital_multiples = []
max_3 = 0
max_4 = 0
for i in reversed(range(5000,10000)):
    s = str(i)+str(2*i)
    if i <= max_4:
        break
    if isPandigital(s):
        max_4 = i

for i in reversed(range(192,342)):
    s = str(i)+str(2*i)+str(3*i)
    if i <= max_3:
        break
    if isPandigital(s):
        max_3 = i

print(max(max_3, max_4))

File: C++/038.cpp
#include "Utils.h"

using namespace std;

int main() {
  long long int max3 = 0, max4 = 0;

  for (int i=9999; i>=5000; i--) {
    if (i<=max4)
      break;
    long long int num = i*100000 + 2*i;
    if (isPandigital(num))
      max4 = i;
  }

  for (int i=333; i>=192; i--) {
    if (i <= max3)
      break;
    long long int num = i*1000000 + 2*i*1000 + 3*i;
    if (isPandigital(num))
      max3 = i;
  }

  if (max3 > max4)
    cout << max3*1000000 + 2*max3*1000 + 3*max3;
  else
    cout << max4*1000000 + 2*max4;
}
